internal/scheduler/adapters.go

```diff
@@ -22,3 +24,31 @@ func (j *JobRepositoryAdapter) TryLeaseJobs(clusterId string, queue string, jobs
 func NewJobRepositoryAdapter(jobRepo repository.JobRepository) *JobRepositoryAdapter {
 	return &JobRepositoryAdapter{jobRepository: jobRepo}
 }
+
+type JobDbAdapter struct {
+	txn *memdb.Txn
+}
+
+func NewJobDbAdapter(txn *memdb.Txn) *JobDbAdapter {
+	return &JobDbAdapter{txn}
+}
+
+func (j *JobDbAdapter) GetJobIterator(_ context.Context, queue string) (JobIterator[*SchedulerJob], error) {
```

Comment:
Unnecessary type parameter?

Reply:
I think it is needed- at least until we can remove the need for the scheduling report to contain a concrete job